UX improvements change status dialog

PVince81 opened the pull request:
### Description

Focus on the custom message field after picking an emoji.
Enter key in the custom message field saves the status.
~~Disable the save button when no custom message is entered (would result
in a server error).~~
Disable save buttons while saving is in progress.

Fixes setting an empty message for clearing when not touching the field (https://github.com/nextcloud/server/issues/23411)

### Motivation
~~
This was mostly because I noticed that entering no custom message triggered an error. I didn't manage to make the server accept empty messages (and even then who knows what side effects...), so I went with prevent empty status message.
And as a bonus added the enter key + autofocus.
~~

### Issues

Partial fix for https://github.com/nextcloud/server/issues/23276 by introducing "save by enter key"
Fixes https://github.com/nextcloud/server/issues/23411

PVince81 committed:
UX improvements change status dialog

Focus on the custom message field after picking an emoji.
Hitting the enter key while in the custom message field now triggers
saving.
Disable save buttons while saving is in progress.

Signed-off-by: Vincent Petry <vincent@nextcloud.com>

PVince81 commented:
Regarding compiled assets, I had to guess by looking at other PRs.
I've updated the README to make it clearer: https://github.com/nextcloud/server/pull/23306

nickvergessen commented on apps/user_status/src/components/SetStatusModal.vue:
```diff
@@ -49,19 +49,21 @@
 					{{ $t('user_status', 'Clear status message') }}
 				</button>
-				<button class="status-buttons__primary primary" @click="saveStatus">
+				<button class="status-buttons__primary primary" :disabled="isSavingStatus || !message" @click="saveStatus">
```
```suggestion
				<button class="status-buttons__primary primary" :disabled="isSavingStatus" @click="saveStatus">
```

Setting to empty clears the status just fine, might be less confusing

PVince81 replied:
It didn't in my instance and caused a server error as it was sending a status icon + empty message.
I personally wanted to set a status without message at that time, this is how I found the issue. Another user might think likewise. So if we fix it to clear the status on empty, the icon would not be set and would not match the user's expectation here.

nickvergessen replied:
https://github.com/nextcloud/server/pull/23108

PVince81 replied:
ok, then it's either a regression or I found an action path that doesn't work:
<img width="862" alt="image" src="https://user-images.githubusercontent.com/277525/95841951-3f8cdc80-0d46-11eb-8a9e-d1b3a2ebe1c7.png">

I'll split this into a separate PR and keep this PR here for just the enter key improvements

PVince81 commented:
I've now removed the code related to empty messages.

The disabled attribute is still there but only for disabling the button when saving as we do in other places.

Adjusted and squashed.

PVince81 commented:
I've raised https://github.com/nextcloud/server/issues/23411 for the empty status issue

PVince81 commented:
I did rebuild and check in the assets. Strange.

After rerunning `make dev-setup` from scratch the build marked those files again as modified.
Anyway, I've pushed them now.

PVince81 committed:
Fix status clearing with initial empty value

When selecting an emoji but not specifying a message, the internal
message value was null which caused a server validation error on
sending.

This fixes the SetStatusModal to always work with an empty string
upfront, as this is the value the field would have if edited and cleared
manually.

Signed-off-by: Vincent Petry <vincent@nextcloud.com>

PVince81 commented:
ouch, pushed on wrong branch... fixing...

PVince81 commented:
okay, I decided to push the empty message fix here as well to avoid needless future conflicts.

Fixes https://github.com/nextcloud/server/issues/23411

jancborchardt reviewed (approved):
Good stuff design-wise, great work @PVince81! :)

PVince81 commented:
there was a weird unrelated error:
```
1) Test\Files\Cache\ScannerTest::testETagRecreation
Failed asserting that null is of type "string".
```

build restarted

PVince81 commented:
@nickvergessen CI green, can you review teh codez ?

rullzer commented:
@PVince81 could you rebase so we can get this in?

PVince81 commented:
Rebased and rebuilt the module in question, as it was the only conflict.
